ui/src/main/webapp/src/app/components/reports/application-details/application-details.component.ts

```diff
@@ -136,22 +151,42 @@ export class ApplicationDetailsComponent implements OnInit {
         });
     }
 
-    private cacheTagsForFile(file:PersistedTraversalChildFileModel, taggableModel:TaggableModel) {
-        taggableModel.tagModel.subscribe((tagModel) => {
-            if (tagModel == null)
-                return;
+    private filterAndSetFiles(traversal:PersistedProjectModelTraversalModel, files:PersistedTraversalChildFileModel[]) {
+        let tagFilterService = new TagFilterService(this.group.reportFilter);
+        files = files.filter((file) => {
+            let tags = this.tagsByFile.get(file.vertexId);
+            if (!tags)
+                return true;
+
+            for (let tag of tags) {
+                if (!tagFilterService.tagMatches(tag)) {
```

Comment:
I'm going to rewrite that too. I think the logic should only exclude the hints or classifications that don't match, not the whole file.